addons/udes_stock/models/stock_picking.py

```diff
@@ -2598,6 +2598,71 @@ def search_for_pickings(self, picking_type_id, picking_priorities, limit=1, doma
 
         return picking
 
+    def _find_unfulfillable_moves(self, batch_size=1000):
+        """Find unfullfilable order lines due to lack of stock."""
+        # This method is patterned on SaleOrder._find_unfulfillable_order_lines
+        # in udes_sale_stock.  Changes applied there should be considered for
+        # this method too.
+        Location = self.env["stock.location"]
+        MoveLine = self.env["stock.move"]
+        Quant = self.env["stock.quant"]
+
+        # Create empty record sets for SO lines
+        unfulfillable_lines = MoveLine.browse()
+
+        _logger.info("Checking pickings to cancel due to stock shortage")
+        # Get unreserved stock for each product in locations
+        locations = Location.get_available_stock_locations()
+        stock = defaultdict(int)
+        skip_states = ("assigned", "done", "cancel")
+
+        for r, batch in self.batched(size=batch_size):
+            _logger.info("Checking pickings %d-%d", r[0], r[-1])
```

Comment:
Can we include the total amount of pickings, i.e. `Checking pickings x-y, out of z` so we have an idea how far along we are.

Reply:
Or a message to state how many we are about to do before starting.